src/cli/commands/pack.js

```diff
@@ -68,14 +68,17 @@ function addEntry(packer: any, entry: Object, buffer?: ?Buffer): Promise<void> {
 
 export async function pack(config: Config, dir: string): Promise<stream$Duplex> {
   const pkg = await config.readRootManifest();
-  const {bundledDependencies, files: onlyFiles} = pkg;
+  const {bundledDependencies, main, files: onlyFiles} = pkg;
 
   // inlude required files
   let filters: Array<IgnoreFilter> = NEVER_IGNORE.slice();
   // include default filters unless `files` is used
   if (!onlyFiles) {
     filters = filters.concat(DEFAULT_IGNORE);
   }
+  if (main) {
```

Comment:
sure, @josephfrazier.
You've added a new condition checking for `main` in package.json.
But have not added a test that covers this condition.
In `__tests__/fixtures/pack/files-include-mandatory/package.json` you probably wanted to add `main` field with `index.js` and not have it in `files`?